### DISCUSSION TITLE
Please up-vote to fix data portal methods being faded out - IDE0051

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#4324] [General]
In a recent Visual Studio update, our beloved data portal methods are being faded out because of `IDE0051 Private member '[method name]' is unused`. Microsoft has not provided an option to disable this "feature".

Would you mind taking a moment to up-vote this issue on the Developer Community so we get this prioritised:

https://developercommunity.visualstudio.com/t/unused-private-member-IDE0051-fading-e/10781612

**Comment by mtavares628** [comment]
It looks like they closed that issue as a duplicate.  The related issue is here:

https://developercommunity.visualstudio.com/t/Private-Unity-messages-incorrectly-marke/10779025

**Reply by michaelcsikos** [reply]
Yes, extremely frustrating, as it's clearly not the same issue. They should reopen it.

**Reply by Carpenter360** [reply]
Hi @michaelcsikos,

I'm assuming you know about the SuppressMessage attribute but just in case. 

[Insert]
**[SuppressMessage("Style", "IDE0051:Remove unused private members", Justification = "Used by CSLA data portal")]**
private void Insert([Inject] IReceiptReaderRequestEditDal dal, [Inject] IMapper mapper)


Cheers,

Sylvain

**Reply by michaelcsikos** [reply]
Hi @Carpenter360, thanks, yes, I know we can switch it off per method or, even better, per assembly with a global suppression `[assembly: SuppressMessage(...)]`. This can be set up in a `cs` file and linked to each business project, but we have dozens of them, and I don't mind the warning itself — it's the fading that is annoying and there's currently no way to stop it without suppressing the warning entirely.

**Reply by ajohnstone-ks** [reply]
> [Insert] **[SuppressMessage("Style", "IDE0051:Remove unused private members", Justification = "Used by CSLA data portal")]** private void Insert([Inject] IReceiptReaderRequestEditDal dal, [Inject] IMapper mapper)

This is what we do, and it doesn't dim the code.  But we also have R#, so maybe that is fixing the issue?

**Reply by michaelcsikos** [reply]
@ajohnstone-ks, `[SuppressMessage]` works for C#, too, but you have to apply the attribute to each method, or globally. I would prefer a Visual Studio option to simply disable the dimming.

**Comment by rockfordlhotka** [comment]
This is one part of a bigger problem, which includes how VS grays out text, but also how AOT works. I have some thoughts on a solution.

#4359

**Comment by Chicagoan2016** [comment]
Chris Wang (MSFT) recommended to add this to editorconfig file
[*.{cs,vb}]
dotnet_diagnostic.IDE0051.severity = none

I tried and it works but did I break something else? :)

**Comment by michaelcsikos** [comment]
@Chicagoan2016, I think the easiest fix for now is to set up a `GlobalSuppressions` folder with the following:
```cs
// This file is used by Code Analysis to maintain SuppressMessage
// attributes that are applied to this project.
// Project-level suppressions either have no target or are given
// a specific target and scoped to a namespace, type, member, etc.

using System.Diagnostics.CodeAnalysis;

[assembly: SuppressMessage("Style", "IDE0051:Remove unused private members")]
```
`SuppressIDE0051.cs` file in this zip:
[SuppressIDE0051.zip](https://github.com/user-attachments/files/18014832/SuppressIDE0051.zip)
